### PR TITLE
AP-2935 Add GOVUK Rubocop Gem

### PR DESCRIPTION
**Pull request by mpw5**
## What

[Link to story](https://dsdmoj.atlassian.net/browse/AP-2935)

Replaces the generic rubocop gem with the GOVUK version to ensure we meet cross-gov code standards and have consistency between our applications.

* Adds all offences to a (very long) `.rubocop_todo.yml` file so that they can be addressed later.

* Where we have previously overriden cops in `.rubocop.yml`, I have left the configuration in if it targets specific file/s (as we know we have unresolvable issues with some classes) but removed it if it was a generic exclusion, as we should now be aiming to conform to the GOVUK standard.

## Checklist

Before you ask people to review this PR:

- Tests and rubocop should be passing: `bundle exec rake`
- Github should not be reporting conflicts; you should have recently run `git rebase main`.
- There should be no unnecessary whitespace changes. These make diffs harder to read and conflicts more likely.
- The PR description should say what you changed and why, with a link to the JIRA story.
- You should have looked at the diff against main and ensured that nothing unexpected is included in your changes.
- You should have checked that the commit messages say why the change was made.

### CODE REVIEW

File: .rubocop.yml [hunk 1]
```diff
@@ -1,9 +1,18 @@
...
+    - config/default.yml
+    - config/rails.yml
+    - config/rspec.yml
+
+inherit_mode:
+  merge:
+    - Exclude
+
 AllCops:
   TargetRubyVersion: 2.7
```
Comment (colinbruce) [inline .rubocop.yml:17]: Should we have updated this to 3.x? 🤔

**Review by colinbruce — APPROVED** [review]
Looks good (apart from something _I_ may have missed) :+1:

File: .rubocop.yml (hunk 1 above)
Comment (mpw5) [reply .rubocop.yml:17]: Good spot, I completely missed that. I'll remove it, if it's not present then rubocop falls back on `.ruby-version`. There doesn't seem to be any benefit in configuring it in multiple places 🤷

Comment (colinbruce) [reply .rubocop.yml:17]: Me too - I've updated ruby multiple times and not spotted it 😬 
Thanks!

**Pull request merged**